src/MensaWidget/MWFrame.class.st

```diff
@@ -40,94 +41,105 @@ MWFrame class >> contentExtent [
+MWFrame class >> scrollBarThickness [
 
-	^ 5 @ 30
+	^ 14 * RealEstateAgent scaleFactor 
 ]
 
 {
 	#category : #constants,
-	#'squeak_changestamp' : 'Oscar 6/5/2024 01:01'
+	#'squeak_changestamp' : 'Louis Christ 6/18/2024 14:45'
 }
-MWFrame class >> scrollBarThickness [
+MWFrame class >> windowName [
 
-	^ 20
+	^ 'Mensa Widget'
 ]
 
 {
 	#category : #dayManipulation,
-	#'squeak_changestamp' : 'Oscar 6/16/2024 23:56'
+	#'squeak_changestamp' : 'Louis Christ 6/18/2024 15:34'
 }
-MWFrame >> addMeals: aDate [
-
-	| contentMorph |
+MWFrame >> abandonContent [
 	
-	contentMorph := self buildContentMorph.
-	(MWDataGettingWrapper getMealsOnDate: aDate yyyymmdd) do: 
-		[:each | contentMorph addMorph: each getCanteenMorph].
-	self contentMorph scroller 
-		layoutPolicy: TableLayout new;
-		addMorph: contentMorph
+	self contentMorph submorphs do: [:each | each abandon]
 
 	
 ]
 
 {
 	#category : #creation,
-	#'squeak_changestamp' : 'Oscar 6/16/2024 23:52'
+	#'squeak_changestamp' : 'Louis Christ 6/18/2024 17:40'
 }
-MWFrame >> buildContentMorph [
+MWFrame >> addContentMorph [
 
-	^ AlignmentMorph newColumn 
+	self contentMorph: (AlignmentMorph newColumn 
 		 cellInset: self class cellSpacing;
 		 extent: self class contentExtent;
-		 hResizing: #spaceFill;
-		 vResizing: #spaceFill
+		vResizing: #shrinkWrap;
+		yourself).
+	self scroller addMorph: self contentMorph
 ]
 
 {
-	#category : #creation,
-	#'squeak_changestamp' : 'Oscar 6/16/2024 23:51'
+	#category : #dayManipulation,
+	#'squeak_changestamp' : 'Louis Christ 6/18/2024 16:05'
 }
-MWFrame >> buildLayout [
+MWFrame >> addMeals: aDate forTime: aTime [
 
-	^ AlignmentMorph newColumn
-		hResizing: #spaceFill;
-		vResizing: #spaceFill;
-		extent: self class contentExtent;
-		color: MWColorTheme backgroundColor;
-		addMorph: self buildScrollPane;
-		addMorph: self buildMenuBar;
-		yourself
+	| morph |
+	
+	(MWDataGettingWrapper getMealsOnDate: aDate yyyymmdd) do: 
+		[:each | morph := (each buildForTime: aTime).
+		self contentMorph addMorph: morph.
+		morph vResizing: #shrinkWrap].
+
+	
 ]
 
 {
 	#category : #creation,
-	#'squeak_changestamp' : 'Louis Christ 6/10/2024 17:59'
+	#'squeak_changestamp' : 'Louis Christ 6/18/2024 14:53'
 }
-MWFrame >> buildMenuBar [
+MWFrame >> addMenuBar [
 	
-	^ MWMenuBar new target: self
+	self menuBar: (MWMenuBar new target: self);
+		addMorph: self menuBar
 ]
 
 {
 	#category : #creation,
-	#'squeak_changestamp' : 'Oscar 6/16/2024 23:58'
+	#'squeak_changestamp' : 'Louis Christ 6/18/2024 17:48'
 }
-MWFrame >> buildScrollPane [
+MWFrame >> addScrollPane [
 
-	^ self contentMorph: 
+	self scroller: 
 		(ScrollPane new 
 			scrollBarThickness: self class scrollBarThickness;
-			color: MWColorTheme backgroundColor;
-			extent: self class contentExtent;
+			color: MWColorTheme defaultColor;
 			hResizing: #spaceFill;
 			vResizing: #spaceFill;
-			yourself);
-		contentMorph
+			scroller);
+		addMorph: self scroller owner.
+	self scroller layoutPolicy: TableLayout new.
```

Comment:
I think the line can be added to the cascade.

Reply:
I dont think it can because there is more than one folloing method call